data_store/validation/towns_fund/fund_specific_validation/fs_validate_r6.py

```diff
@@ -0,0 +1,59 @@
+from datetime import datetime
+
+import pandas as pd
+
+from data_store.const import StatusEnum
+from data_store.messaging.tf_messaging import TFMessages as msgs
+from data_store.validation.towns_fund.failures.user import GenericFailure
+from data_store.validation.towns_fund.fund_specific_validation.fs_validate_r4 import (
+    validate as validate_r4,
+)
+from data_store.validation.towns_fund.fund_specific_validation.fs_validate_r4 import (
+    validate_project_progress as validate_project_progress_r4,
+)
+
+
+def validate(
+    data_dict: dict[str, pd.DataFrame], original_workbook: dict[str, pd.DataFrame] | None = None
+) -> list[GenericFailure]:
+    """Top-level Towns Fund Round 6 specific validation."""
+
+    # Start with validations from Round 4
+    validation_failures = validate_r4(data_dict, original_workbook)
```

Comment:
This will be tying together the R4 and R6 validation very closely and might make it difficult if we want to 'retire' R4 code at some point. I guess a similar concern as we had for R5, and how much we want to separate things out and make them each their own mini pipeline, especially when they're so similar (it seems silly to add it for one additional check).

This is more a general point in case it's something worth discussing as a team.

Reply:
Yeah very fair point, as we did this for R4 and R5 I don't want to rock the boat too much on this specific PR but definitely worth considering whether we want to refactor, and how far we want to isolate code between rounds etc.